react/lesson1.md

```diff
@@ -36,175 +91,431 @@ cd my-app/
-                name= {organisation.name}
-                borough={organisation.borough ? organisation.borough : 'None'}
-                website= {organisation.website}
-            />
-        )
+```javascript
+  difficultyChange = (event) => {
+    this.setState({
+      difficulty: event.target.value
     });
+  };
+
+  buildGame() {
+    let buttons =  [];
+    for(var i=0; i<this.state.difficulty; i++) {
+      buttons.push(<GameButton />)
+    }
+    return buttons;
+  }
+
+  <input value={this.state.difficulty} type="number" onChange={this.difficultyChange}/>
+```
+
+Notice I'm also passing the state difficulty to our input through the `value` attribute.
+
+Now go ahead and change the difficulty level in your input, and see the number of buttons update in real time. Try to keep it under 100 to avoid your browser freezing.
+
+# Props
+
+Remember HTML attributes? 
+
+`<input value="Hervin" class="nameInput" />`
+
+Props are the inputs sent from the parent to child component via its attributes. So for our `GameButton` we could send it a prop that would specify what text it should display on the button:
+
+`<GameButton btnText="First button" />`
+
+But how do we access this "btnText" attribute internally and display it?
+
+```javascript
+function GameButton(props) {
+    return <button> {props.btnText} </button>;
 }
-class App extends Component {
+```
+
+All the props get passed as properties of the "props" object, as the first argument of our component function.
+
+For class components, props can be accessed via the `this.props` object:
+
+```javascript
+class GameButton extends React.Component {
   render() {
-      return (
-        <div className="App">
-          {this.renderOrganisations()}
-        </div>
-      );
+    return <button>{this.props.btnText}</button>
   }
 }
+```
+
+### Let's pass some props in our game!
+
+Add a selected attribute to the button:
 
+```javascript
+<GameButton selected={false}/>
 ```
 
-- [Mapping Exercise](https://codepen.io/kabaros/pen/BpLzzE?editors=0010#0)
+Then let's give a different css class to our button depending on whether it's selected or not:
 
-State
----
-Props are a Component's configuration, they are received from above and they don't usually change by the component itself.
-State starts with a default value (set in the constructor typically) then it gets changed (mutated) by events of the application (user interactions, form input, API calls etc..)
+```javascript
+import './gameButton.css';
 
+function GameButton(props) {
+    let className = "neutral";
+    if(props.selected === true) {
+        className = "selected";
+    }
+    return <button className={className}> Awesome button! </button>;
+}
+```
+
+Create the `gameButton.css` file and defined your rules:
+
+```css
+.neutral {
+    background: lightgrey;
+}
+
+.selected {
+    background: red;
+}
+```
+
+Now try changing the `selected` attribute on your `GameButton`. It should change color depending on whether it's true or false.
```

Comment:
I haven't seen where you have mentioned why we need to use `className` over `class`.

(Apologies if i've missed it - but it's likely to be a question that comes up)

Reply:
I mentioned it in the JSX gotchas. Thanks for checking, though!